TestSwitchApi/readme.md

```diff
@@ -0,0 +1,13 @@
+# TestSwitch API
+This is the API for the TestSwitch.
+
+### Linting
+We're using StyleCop to lint the project.  
+Some StyleCop rules are disabled in the .ruleset files in each project. 
+ 
+Compile-time warnings are set to be treated as errors in the csproj files.  
+This is to ensure Circle CI fails the build if there are lint errors.  
+This can be changed during dev if it's a pain by setting the value of `TreatWarningsAsErrors` to false.
+
+I'd recommend using auto-format (`ctrl+shift+L` is the default Rider command, or go to Code > Reformat Code) to auto-format files if StyleCop is complaining about formatting.  
+If Rider disagrees with StyleCop as to how to format code, make sure that Rider's code style settings match StyleCop's settings (Settings (`ctrl+shift+s`) > Editor > Code Style > C#).
```

Comment:
It sounds like it might be possible to get Rider to read your stylecop settings - https://www.jetbrains.com/help/rider/StyleCop_Styles.html

Could you see if that works?
If so - might be a nice way to make dev nice and easy

Reply:
Yeah, it seems like settings.stylecop only works with the old StyleCope package which is a bit out of date (seems to go up to C#6 just).

I've added a DotSettings file which should share code style and enable Roslyn StyleCop analyzers. https://www.jetbrains.com/help/rider/Using_NET_Compiler_Analyzers.html